packages/trpc/server/routers/loggedInViewer/integrations.handler.ts

```diff
@@ -205,46 +78,15 @@ export const integrationsHandler = async ({ ctx, input }: IntegrationsOptions) =
         isSetupAlready,
         ...(app.dependencies && { dependencyData }),
       };
+
+      return appData;
     })
   );
 
-  if (variant) {
-    // `flatMap()` these work like `.filter()` but infers the types correctly
-    apps = apps
-      // variant check
-      .flatMap((item) => (item.variant.startsWith(variant) ? [item] : []));
-  }
-
-  if (exclude) {
-    // exclusion filter
-    apps = apps.filter((item) => (exclude ? !exclude.includes(item.variant) : true));
-  }
-
-  if (onlyInstalled) {
-    apps = apps.flatMap((item) =>
-      item.userCredentialIds.length > 0 || item.teams.length || item.isGlobal ? [item] : []
-    );
-  }
-
-  if (extendsFeature) {
-    apps = apps
-      .filter((app) => app.extendsFeature?.includes(extendsFeature))
-      .map((app) => ({
-        ...app,
-        isInstalled: !!app.userCredentialIds?.length || !!app.teams?.length || app.isGlobal,
-      }));
-  }
-
-  if (sortByMostPopular) {
-    const installCountPerApp = await getInstallCountPerApp();
-
-    // sort the apps array by the most popular apps
-    apps.sort((a, b) => {
-      const aCount = installCountPerApp[a.slug] || 0;
-      const bCount = installCountPerApp[b.slug] || 0;
-      return bCount - aCount;
-    });
-  }
+  apps = await transformAppsBasedOnInput({
+    apps,
+    input: { exclude, extendsFeature, onlyInstalled, sortByMostPopular, variant },
+  });
```

Comment:
Good candidate for the [presenter pattern](https://github.com/nikolovlazar/nextjs-clean-architecture/blob/9b9667ca3f745aa11164268e7cea3d2bcb1212fd/src/interface-adapters/controllers/todos/get-todos-for-user.controller.ts#L8-L17)